test/integ/conftest.py

```diff
@@ -1,87 +1,281 @@
+        raise Exception(
+            "The environment variable WORKER_REGION is no longer supported. Please use REGION instead."
+        )
+
+    # Prepare the Worker agent Python package
+    worker_agent_whl_path = os.getenv("WORKER_AGENT_WHL_PATH")
+    if worker_agent_whl_path:
+        LOG.info(f"Using Worker agent whl file: {worker_agent_whl_path}")
+        resolved_whl_paths = glob.glob(worker_agent_whl_path)
+        assert (
+            len(resolved_whl_paths) == 1
+        ), f"Expected exactly one Worker agent whl path, but got {resolved_whl_paths} (from pattern {worker_agent_whl_path})"
+        resolved_whl_path = resolved_whl_paths[0]
+
+        dest_path = posixpath.join("/tmp", os.path.basename(resolved_whl_path))
+        file_mappings = [(resolved_whl_path, dest_path)]
+
+        LOG.info(f"The whl file will be copied to {dest_path} on the Worker environment")
+        worker_agent_requirement_specifier = dest_path
+    else:
+        worker_agent_requirement_specifier = os.getenv(
+            "WORKER_AGENT_REQUIREMENT_SPECIFIER",
+            "deadline-cloud-worker-agent",
+        )
+        LOG.info(f"Using Worker agent package {worker_agent_requirement_specifier}")
+
+    # Path map the service model
+    with tempfile.TemporaryDirectory() as tmpdir:
+        src_path = pathlib.Path(tmpdir) / f"{service_model.service_name}-service-2.json"
+
+        LOG.info(f"Staging service model to {src_path} for uploading to S3")
+        with src_path.open(mode="w") as f:
+            json.dump(service_model.model, f)
+
+        dst_path = posixpath.join("/tmp", src_path.name)
+        LOG.info(f"The service model will be copied to {dst_path} on the Worker environment")
+        file_mappings.append((str(src_path), dst_path))
+
+        yield DeadlineWorkerConfiguration(
+            farm_id=deadline_resources.farm.id,
+            fleet_id=deadline_resources.scaling_fleet.id,
+            region=region,
+            user=os.getenv("WORKER_POSIX_USER", "deadline-worker"),
+            group=os.getenv("WORKER_POSIX_SHARED_GROUP", "shared-group"),
+            allow_shutdown=True,
+            worker_agent_install=PipInstall(
+                requirement_specifiers=[worker_agent_requirement_specifier],
+                codeartifact=codeartifact,
+            ),
+            service_model_path=dst_path,
+            file_mappings=file_mappings or None,
+        )
 
 
 @pytest.fixture(scope="session")
-def queue_with_job_run_as_user(
-    farm: Farm,
-    fleet: Fleet,
-    deadline_client: DeadlineClient,
-    job_run_as_user: PosixSessionUser,
-) -> Generator[Queue, None, None]:
-    queue = Queue.create(
-        client=deadline_client,
-        display_name=f"Queue with jobsRunAsUser {job_run_as_user.user}",
-        farm=farm,
-        job_run_as_user=JobRunAsUser(posix=job_run_as_user, runAs="QUEUE_CONFIGURED_USER"),
-    )
+def worker(request: pytest.FixtureRequest, worker_config) -> Generator[DeadlineWorker, None, None]:
```

Comment:
Why are we redefining this fixture that already exists here: https://github.com/aws-deadline/deadline-cloud-test-fixtures/blob/mainline/src/deadline_test_fixtures/fixtures.py#L473

Reply:
This should have been defined here originally using the DeadlineWorker, and DeadlineWorkerConfig classes from deadline-cloud-test-fixtures. Since each repository could have different implementations of the worker and its configuration.